eZ/Publish/Core/Repository/LocationService.php

```diff
@@ -967,4 +942,47 @@ public function find(Filter $filter, ?array $languages = null): LocationList
             ]
         );
     }
+
+    private function checkCreatePermissionOnSubtreeTarget(APILocation $targetParentLocation, APILocation $loadedSubtree, APILocation $loadedTargetLocation): void
+    {
+        $locationTarget = (new DestinationLocationTarget($targetParentLocation->id, $loadedSubtree->contentInfo));
+        if (!$this->permissionResolver->canUser(
+            'content',
+            'create',
+            $loadedSubtree->getContentInfo(),
+            [$loadedTargetLocation, $locationTarget]
+        )) {
+            throw new UnauthorizedException(
+                'content',
+                'create',
+                ['locationId' => $loadedTargetLocation->id]
+            );
+        }
+
+        // Check read access to whole source subtree
+        $contentReadCriterion = $this->permissionCriterionResolver->getPermissionsCriterion(
+            'content',
+            'read'
+        );
+        if ($contentReadCriterion === false) {
+            throw new UnauthorizedException('content', 'read');
+        } elseif ($contentReadCriterion !== true) {
+            // Query if there are any content in subtree current user don't have access to
+            $query = new Query(
+                [
+                    'limit' => 0,
+                    'filter' => new CriterionLogicalAnd(
+                        [
+                            new CriterionSubtree($loadedSubtree->pathString),
+                            new CriterionLogicalNot($contentReadCriterion),
+                        ]
+                    ),
+                ]
+            );
+            $result = $this->repository->getSearchService()->findContent($query, [], false);
+            if ($result->totalCount > 0) {
+                throw new UnauthorizedException('content', 'read');
+            }
+        }
```

Comment:
Since the original condition results in an exception, this can be split into subsequent checks (instead of `elseif`).

```suggestion
        } 
        
        if ($contentReadCriterion !== true) {
            // Query if there are any content in subtree current user don't have access to
            $query = new Query(
                [
                    'limit' => 0,
                    'filter' => new CriterionLogicalAnd(
                        [
                            new CriterionSubtree($loadedSubtree->pathString),
                            new CriterionLogicalNot($contentReadCriterion),
                        ]
                    ),
                ]
            );
            $result = $this->repository->getSearchService()->findContent($query, [], false);
            if ($result->totalCount > 0) {
                throw new UnauthorizedException('content', 'read');
            }
        }
```

I do think that we should check against `=== null` instead, but there is no difference at the end of the day, as only 3 values are allowed (`null`, `false`, `true`), and `false` cannot be one of them due to previous check.

Reply:
Done, however, this wasn't a part of PR.